### PR TITLE
Fix for #11663

### LINKED ISSUE #11663: Deleting an HTML Anchor attribute leaves an empty HTML id attribute
**Describe the bug**
When you delete a HTML Anchor attribute, it leaves an empty HTML id attribute in the HTML code which is unvalidated markup (W3C validator).

**To Reproduce**
1. Insert a heading block and type some text
2. In the Advanced panel, set an HTML Anchor attribute  
3. Empty the HTML Anchor field
4. Switch to Code Editor. 
````
<!-- wp:heading -->
<h2 id="">Heading</h2>
<!-- /wp:heading -->
````

**Desktop (please complete the following information):**
 - OS: Windows 10, Mac OS
 - Browser: Chrome, Safari Firefox, Edge

**Additional context**
Gutenberg 4.2.0

### PR DESCRIPTION
**Pull request by saleehk**
## Description
Fixed https://github.com/WordPress/gutenberg/issues/11663, 

## How has this been tested?
Checked it manually, There no unit test or end2end test for checking Anchor attribute 

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation.

### CODE REVIEW

**Comment by dmsnell** [comment]
I wonder if it would be more appropriate to perform the filtering of the attribute in `addSaveProps` since that's the point where we're storing this and it will affect the underlying block. also, I think there are some plausible flows that bypass the inspector controls leaving the same bug open.

**Comment by saleehk** [comment]
Updated with addSaveProps

**Review by dmsnell — APPROVED** [review]
This looks like a nice improvement to me - thanks @saleehk 

@aduth do you see any problem with this fix? I'm not sure if there is a good reason to generalize past here in this PR but I'm guessing we have similar issues across many blocks and plugins.

File: packages/editor/src/hooks/anchor.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ export const withInspectorControl = createHigherOrderComponent( ( BlockEdit ) =>
  */
 export function addSaveProps( extraProps, blockType, attributes ) {
 	if ( hasBlockSupport( blockType, 'anchor' ) ) {
-		extraProps.id = attributes.anchor;
+		extraProps.id = attributes.anchor === '' ? null : attributes.anchor;
```
Comment (aduth) [inline packages/editor/src/hooks/anchor.js]: Alternatively, could this have been considered as part of the condition for assigning `id`?

i.e.

```js
if ( hasBlockSupport( blockType, 'anchor' ) && attributes.anchor ) {
```

**Pull request merged**